### PR TITLE
TSCH with Rime

### PR DESCRIPTION
**Pull request by simonduq**
This PR makes TSCH work with short addresses and Rime. Added new example for Rime/TSCH. Useful for testing TSCH without a full IPv6 stack.

Note that this PR includes https://github.com/contiki-os/contiki/pull/1609

### CODE REVIEW

**Comment by alignan** [comment]
We need to wait for https://github.com/contiki-os/contiki/pull/1609 to be merged, right?

**Comment by simonduq** [comment]
I think that would be cleaner, yes. The idea was to give a chance to non-tsch experts to review the non-tsch related part first (https://github.com/contiki-os/contiki/pull/1609)

**Comment by pablocorbalan** [comment]
I have tried the code of this PR and it works nicely. I tested it also with the reliable unicast (runicast) primitive and Contiki Collect and it seems to work fine with all of them. Nice work! 👍 

I may test this a bit more extensively (perhaps, even in some testbed). I may also try to create a simple scheduler for Contiki Collect + TSCH. I'll let you know my results!

One question: are the changes of commit [0c79d23](https://github.com/contiki-os/contiki/pull/1610/commits/0c79d23ed6cf5e0284bbfa0f1f4383c75da8e772) necessary? The functions that were used depend on `nbr-table.[ch]` but not on IP. I feel that we may make the code a bit more complex just to avoid using the `nbr-table`module. Also, if we would eventually update Contiki Collect to use the `nbr-table`module, perhaps, it would make more sense to keep using that module here too.

File: examples/rime-tsch/node.c [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  while(1) {
+    static struct etimer et;
+
+    etimer_set(&et, CLOCK_SECOND);
+
+    PROCESS_WAIT_EVENT_UNTIL(etimer_expired(&et));
+
+    packetbuf_copyfrom("Hello", 5);
+
+    if(linkaddr_cmp(&sender_addr, &linkaddr_node_addr)) {
```
Comment (pablocorbalan) [inline examples/rime-tsch/node.c]: You could change this line to:
`if(!linkaddr_cmp(&destination_addr, &linkaddr_node_addr)) {`
and then remove the definition of `sender_addr` on top. This way, if someone tries this example with multiple nodes, all the nodes will send unicast messages to the destination.

Comment (simonduq) [inline examples/rime-tsch/node.c]: sure! will do

Comment (simonduq) [inline examples/rime-tsch/node.c]: done

**Comment by simonduq** [comment]
@pablocorbalan good point regarding the neighbor table for time source selection. Cannot seem to recall why I did this. I was assuming nbr-table was IP-only at some point but this is indeed not true. Will try to look into this again (or if you happen to do, as you mentioned you might use this, please share your conclusions!)

**Comment by simonduq** [comment]
I did as suggested by @pablocorbalan, that is, revert back to using nbr-table for eb-autoselect. A few fixes were needed to get it to work with 2-byte addresses and with other recent changes. This PR is ready IMO (the diff has become very small!)

**Comment by pablocorbalan** [comment]
@simonduq Sorry for the delay! I haven't really found the time to work more on this, so I have no updates. I might push in the future more examples to the `rime-tsch`folder.

I tried the updated code as well with a `runicast` application and `collect`and it also worked fine. This looks great to me with the last changes and using the `nbr-table` module. Thanks!
👍

**Comment by simonduq** [comment]
setting timeout for merging

**Comment by simonduq** [comment]
merging after timeout

**Comment by sieben** [comment]
It doesn't seem to work with wismote as a target platform: http://pastebin.com/XwxSJcZf

**Comment by simonduq** [comment]
That is right, TSCH is not supported on Wismote. More info about an ongoing porting effort at https://github.com/contiki-os/contiki/issues/1517

**Pull request merged**